src/GovUk.Education.ExploreEducationStatistics.Data.Services/ViewModels/LocationViewModel.cs

```diff
@@ -1,23 +1,25 @@
+#nullable enable
+
 namespace GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels
 {
-    public class LocationViewModel
+    public record LocationViewModel
     {
-        public CodeNameViewModel Country { get; set; }
-        public CodeNameViewModel EnglishDevolvedArea { get; set; }
-        public CodeNameViewModel Institution { get; set; }
-        public CodeNameViewModel LocalAuthority { get; set; }
-        public CodeNameViewModel LocalAuthorityDistrict { get; set; }
-        public CodeNameViewModel LocalEnterprisePartnership { get; set; }
-        public CodeNameViewModel MayoralCombinedAuthority { get; set; }
-        public CodeNameViewModel MultiAcademyTrust { get; set; }
-        public CodeNameViewModel OpportunityArea { get; set; }
-        public CodeNameViewModel ParliamentaryConstituency { get; set; }
-        public CodeNameViewModel Provider { get; set; }
-        public CodeNameViewModel Region { get; set; }
-        public CodeNameViewModel RscRegion { get; set; }
-        public CodeNameViewModel School { get; set; }
-        public CodeNameViewModel Sponsor { get; set; }
-        public CodeNameViewModel Ward { get; set; }
-        public CodeNameViewModel PlanningArea { get; set; }
+        public CodeNameViewModel? Country { get; init; }
```

Comment:
Just double-check that we're not caching this anywhere that might complain if it can't find setters when trying to deserialise :)

Reply:
An example area would be Admin's TableBuilderController, with:

```
        [BlobCache(typeof(DataBlockTableResultCacheKey))]
        private async Task<Either<ActionResult, TableBuilderResultViewModel>> GetReleaseDataBlockResults(
```

Reply:
And I'm guessing that if this had been a problem, then it should have been highlighted in `TableBuilderControllerTests.TableBuilderResultViewModel_SerialiseAndDeserialise()`, so we should be good?

Reply:
Did a check by altering the data and it seems to be fine. Thanks for pointing it out and checking!